kinto/core/storage/testing.py

```diff
@@ -981,6 +981,28 @@ def test_purge_deleted_remove_all_tombstones_by_parent_id(self):
                                                  collection_id=None)
         self.assertEqual(num_removed, 2)
 
+    def test_purge_deleted_removes_timestamps_by_parent_id(self):
+        self.create_record(parent_id='/abc/a', collection_id='c')
+        self.create_record(parent_id='/abc/a', collection_id='c')
+        self.create_record(parent_id='efg', collection_id='c')
+
+        before1 = self.storage.collection_timestamp(parent_id='/abc/a', collection_id='c')
+        # Different parent_id with record.
+        before2 = self.storage.collection_timestamp(parent_id='efg', collection_id='c')
+        # Different parent_id without record.
+        before3 = self.storage.collection_timestamp(parent_id='ijk', collection_id='c')
+
+        self.storage.delete_all(parent_id='/abc/*', collection_id=None, with_deleted=False)
+        self.storage.purge_deleted(parent_id='/abc/*', collection_id=None)
+
+        after1 = self.storage.collection_timestamp(parent_id='/abc/a', collection_id='c')
+        after2 = self.storage.collection_timestamp(parent_id='efg', collection_id='c')
+        after3 = self.storage.collection_timestamp(parent_id='ijk', collection_id='c')
```

Comment:
Are the slashes in `/abc/a` part of why the behavior is different? Can we make them all have slashes?